libraries/botbuilder/src/sharepoint/sharePointActivityHandler.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * @module botbuilder
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    ActivityHandler,
+    InvokeResponse,
+    AceRequest,
+    TurnContext,
+    CardViewResponse,
+    QuickViewResponse,
+    GetPropertyPaneConfigurationResponse,
+    SetPropertyPaneConfigurationResponse,
+    HandleActionResponse,
+} from 'botbuilder-core';
+
+/**
+ * The SharePointActivityHandler is derived from ActivityHandler. It adds support for
+ * the SharePoint specific events and interactions
+ */
+export class SharePointActivityHandler extends ActivityHandler {
+    /**
+     * Invoked when an invoke activity is received from the connector.
+     * Invoke activities can be used to communicate many different things.
+     * * Invoke activities communicate programmatic commands from a client or channel to a bot.
+     *
+     * @param context A strongly-typed context object for this turn
+     * @returns A task that represents the work queued to execute
+     */
+    protected async onInvokeActivity(context: TurnContext): Promise<InvokeResponse> {
+        try {
+            if (!context.activity.name && context.activity.channelId === 'sharepoint') {
+                throw new Error('NotImplemented');
+            } else {
+                switch (context.activity.name) {
+                    case 'cardExtension/getCardView':
+                        return ActivityHandler.createInvokeResponse(
+                            await this.OnSharePointTaskGetCardViewAsync(context, context.activity.value as AceRequest)
+                        );
+
+                    case 'cardExtension/getQuickView':
+                        return ActivityHandler.createInvokeResponse(
+                            await this.OnSharePointTaskGetQuickViewAsync(context, context.activity.value as AceRequest)
```

Comment:
[nit] split parameters to 2 lines to match below formatting

Reply:
I can't :) The formatter (seems like) is based on the length of the string.